Python/chapter03/p02_stack_min/miguelHx.py

```diff
@@ -0,0 +1,248 @@
+"""
+Python version 3.7.0
+3.2 - Stack Min
+How would you design a stack which, in addition to
+push and pop, has a function min which returns the minimum element?
+Push, pop and min should all operate in O(1) time.
+"""
+
+import copy
+import unittest
+import sys
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Generic, TypeVar
+from typing import List, Optional, Iterator
+
+T = TypeVar('T')
+C = TypeVar('C', bound='Comparable')
+
+class Comparable(Generic[T, C]):
+    @abstractmethod
+    def __lt__(self, other: T):
+        pass
+
+@dataclass
+class StackNode(Generic[T, C]):
+    data: T
+    next: 'Optional[StackNode[T, C]]'
+    running_min: T
```

Comment:
```suggestion
class StackNode(Generic[T]):
    data: T
    next: 'Optional[StackNode[T]]'
    running_min: T
```

Reply:
Updated here 6d9c43e99e208cde9e734e24591a721c5b2d5c5a in response to another comment